src/go/k8s/pkg/resources/certmanager/issuer.go

```diff
@@ -76,20 +85,48 @@ func (r *IssuerResource) Ensure(ctx context.Context) error {
 	return nil
 }
 
+var errorUnknownIssuer = errors.New("issuer type unknown, expecting IssuerKind or ClusterIssuerKind")
+
 // Obj returns resource managed client.Object
 func (r *IssuerResource) Obj() (k8sclient.Object, error) {
 	objLabels := labels.ForCluster(r.pandaCluster)
-	issuer := &cmapiv1.Issuer{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: r.Key().Namespace,
-			Name:      r.Key().Name,
-			Labels:    objLabels,
-		},
-		Spec: cmapiv1.IssuerSpec{
+	objectMeta := metav1.ObjectMeta{
+		Name:      r.Key().Name,
+		Namespace: r.Key().Namespace,
+		Labels:    objLabels,
+	}
+
+	var spec cmapiv1.IssuerSpec
+	if r.secretName == "" {
+		spec = cmapiv1.IssuerSpec{
 			IssuerConfig: cmapiv1.IssuerConfig{
 				SelfSigned: &cmapiv1.SelfSignedIssuer{},
 			},
-		},
+		}
+	} else {
+		spec = cmapiv1.IssuerSpec{
+			IssuerConfig: cmapiv1.IssuerConfig{
+				CA: &cmapiv1.CAIssuer{
+					SecretName: r.secretName,
+				},
+			},
+		}
+	}
+
+	var issuer k8sclient.Object
+	switch r.issuerType {
+	case cmapiv1.IssuerKind:
```

Comment:
the commit body doesn't explain what these `issuer-types` are and how to think about them.